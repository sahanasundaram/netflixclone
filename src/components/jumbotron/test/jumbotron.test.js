import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'; // Import this to use Jest DOM matchers
import Jumbotron from '../index'; // Replace with the correct path to your Jumbotron component file

describe('Jumbotron Component', () => {
  it('renders with default props', () => {
    const { container } = render(<Jumbotron />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders children components', () => {
    const { getByText } = render(
      <Jumbotron>
        <Jumbotron.Title>Title</Jumbotron.Title>
        <Jumbotron.SubTitle>Subtitle</Jumbotron.SubTitle>
      </Jumbotron>
    );

    expect(getByText('Title')).toBeInTheDocument();
    expect(getByText('Subtitle')).toBeInTheDocument();
  });

  it('renders with custom styles from Container component', () => {
    const { container } = render(
      <Jumbotron.Container style={{ backgroundColor: 'blue' }}>
        Content
      </Jumbotron.Container>
    );

    expect(container.firstChild).toHaveStyle('background-color: blue');
  });

  it('renders with custom styles from Image component', () => {
    const { container } = render(<Jumbotron.Image src="image.jpg" style={{ width: '100%' }} />);
    const image = container.querySelector('img');

    expect(image).toHaveAttribute('src', 'image.jpg');
    expect(image).toHaveStyle('width: 100%');
  });

  // Add more test cases based on your component features and requirements
});
