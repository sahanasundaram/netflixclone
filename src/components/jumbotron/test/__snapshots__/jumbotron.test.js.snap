// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jumbotron Component renders with default props 1`] = `
<div
  class="jumbotron__Item-sc-1m8aix0-1 gHAorY"
  direction="row"
>
  <div
    class="jumbotron__Inner-sc-1m8aix0-0 iWGevn"
    direction="row"
  />
</div>
`;
